app/src/main/java/com/wut/indoornavigation/data/mesh/MeshImpl.java

```diff
@@ -0,0 +1,227 @@
+            if (stairs.getStart() != stairs.getEnd()) {
+                int endFloor = stairs.getEndfloor();
+                List<Stairs> endFloorStairs = building.getFloors().get(endFloor).getStairs();
+                List<Vertex> endFloorStairsGraphVertices = stairsVerticesDict.get(endFloor);
+
+                int endVertexIndex = -1;
+                for (int j = 0; j < endFloorStairs.size(); j++) {
+                    if (endFloorStairs.get(j).getId() == stairs.getEndid()) {
+                        endVertexIndex = j;
+                        break;
+                    }
+                }
+
+                if (endVertexIndex == -1) {
+                    throw new IllegalStateException("This algorithm is bugged as f*ck.");
+                }
+
+                Vertex startVertex = endFloorStairsGraphVertices.get(i);
+                Vertex endVertex = endFloorStairsGraphVertices.get(endVertexIndex);
+
+                graph.addEdge(startVertex, endVertex, EDGE_ELEVATOR_WEIGHT);
+            }
+        }
+    }
+
+    private Vertex ProcessCell(int x, int y, FloorObject[][] enumMap, int floorNumber, boolean[][] visited, Graph graph) {
+        if (visited[x][y]) {
+            return graph.getVertexByCoordinates(x / 2, y / 2);
+        }
+
+        visited[x][y] = true;
+        if (enumMap[x][y] == FloorObject.SPACE || enumMap[x][y] == FloorObject.DOOR || enumMap[x][y] == FloorObject.STAIRS || enumMap[x][y] == FloorObject.ELEVATOR) {
+            Point coordinates = new Point(x / 2, y / 2, floorNumber);
+            Vertex vertex = new Vertex(idSeed--, coordinates);
+            graph.addVertex(vertex);
+
+            List<Vertex> vertices = null;
+            if (enumMap[x][y] == FloorObject.DOOR) {
+                if (destinationVerticesDict.containsKey(floorNumber)) {
+                    vertices = destinationVerticesDict.get(floorNumber);
+                } else {
+                    vertices = new ArrayList<>();
+                    destinationVerticesDict.put(floorNumber, vertices);
+                }
+            }
+            if (enumMap[x][y] == FloorObject.ELEVATOR) {
+                if (elevatorsVerticesDict.containsKey(floorNumber)) {
+                    vertices = elevatorsVerticesDict.get(floorNumber);
+                } else {
+                    vertices = new ArrayList<>();
+                    elevatorsVerticesDict.put(floorNumber, vertices);
+                }
+            }
+            if (enumMap[x][y] == FloorObject.STAIRS) {
+                if (stairsVerticesDict.containsKey(floorNumber)) {
+                    vertices = stairsVerticesDict.get(floorNumber);
+                } else {
+                    vertices = new ArrayList<>();
+                    stairsVerticesDict.put(floorNumber, vertices);
+                }
+            }
+
+            if (vertices != null) {
+                vertices.add(vertex);
+            }
+
+            return vertex;
+        }
+
+        return null;
```

Comment:
czemu zwracasz nulle jak sie nie udalo. Skoro nie udalo sie to throw excetpion jakis. Unikajmy nulli